Octokit.Tests/Clients/MigrationsClientTests.cs

```diff
@@ -0,0 +1,187 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Threading.Tasks;
+using NSubstitute;
+using Xunit;
+
+namespace Octokit.Tests.Clients
+{
+    public class MigrationsClientTests
+    {
+        public class TheGetMethod
+        {
+            [Fact]
+            public void RequestsCorrectUrl()
+            {
+                var connection = Substitute.For<IApiConnection>();
+                var client = new MigrationsClient(connection);
+
+                client.Get("fake", 69);
+
+                connection.Received().Get<Migration>(
+                    Arg.Is<Uri>(u => u.ToString() == "orgs/fake/migrations/69"));
+            }
+
+            [Fact]
+            public async Task EnsuresNonNullAndNonEmptyArguments()
+            {
+                var connection = Substitute.For<IApiConnection>();
+                var client = new MigrationsClient(connection);
+
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.Get(null, 69));
+                await Assert.ThrowsAsync<ArgumentException>(() => client.Get("", 69));
+            }
+        }
+
+        public class TheGetAllMethod
+        {
+            [Fact]
+            public void RequestsCorrectUrl()
+            {
+                var connection = Substitute.For<IApiConnection>();
+                var client = new MigrationsClient(connection);
+
+                client.GetAll("fake");
+
+                connection.Received().Get<List<Migration>>(
+                    Arg.Is<Uri>(u => u.ToString() == "orgs/fake/migrations"));
+            }
+
+            [Fact]
+            public async Task EnsuresNonNullAndNonEmptyArguments()
+            {
+                var connection = Substitute.For<IApiConnection>();
+                var client = new MigrationsClient(connection);
+
+                await Assert.ThrowsAsync<ArgumentNullException>(() => client.GetAll(null));
+                await Assert.ThrowsAsync<ArgumentException>(() => client.GetAll(""));
+            }
+        }
+
+        public class TheStartNewMethod
+        {
+            [Fact]
+            public void RequestsCorrectUrl()
+            {
+                var connection = Substitute.For<IApiConnection>();
+                var client = new MigrationsClient(connection);
+                var migrationRequest = new StartMigrationRequest(new List<string> { "fake/repo" });
+
+                client.Start("fake", migrationRequest);
+
+                connection.Received().Post<Migration>(
+                    Arg.Is<Uri>(u => u.ToString() == "orgs/fake/migrations"),
+                    migrationRequest);
```

Comment:
Minor nitpick but since this test's purpose is to assert the correct Url is hit, you shouldnt "also" verify the request body was passed (there is another test for that later).  So you can use Arg.IsAny<StartMigrationRequest>()` here